solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -529,7 +529,7 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
       }
     } else {
       // a distributed request
-
+      long maxTimeAllowed = req.getParams().getLong(CommonParams.TIME_ALLOWED, Long.MAX_VALUE);
```

Comment:
It looks like we're not actually doing anything with `maxTimeAllowed` here? Probably want to pass it to shardHandler around line 596?